gef-ui/server/controllers/amendments/cover-end-date/get-cover-end-date.test.ts

```diff
@@ -0,0 +1,135 @@
+import * as dtfsCommon from '@ukef/dtfs2-common';
+import { aPortalSessionUser, DEAL_STATUS, PORTAL_LOGIN_STATUS, DEAL_SUBMISSION_TYPE, ROLES } from '@ukef/dtfs2-common';
+import { HttpStatusCode } from 'axios';
+import { createMocks } from 'node-mocks-http';
+
+/* eslint-disable import/first */
+const getApplicationMock = jest.fn();
+const getFacilityMock = jest.fn();
+
+import { getCoverEndDate, GetCoverEndDateRequest } from './get-cover-end-date';
+import { CoverEndDateViewModel } from '../../../types/view-models/amendments/cover-end-date-view-model';
+import { MOCK_BASIC_DEAL } from '../../../utils/mocks/mock-applications';
+import { MOCK_UNISSUED_FACILITY, MOCK_ISSUED_FACILITY } from '../../../utils/mocks/mock-facilities';
+
+jest.mock('../../../services/api', () => ({
+  getApplication: getApplicationMock,
+  getFacility: getFacilityMock,
+}));
+
+const dealId = 'dealId';
+const facilityId = 'facilityId';
+const amendmentId = 'amendmentId';
+
+const getHttpMocks = () =>
+  createMocks<GetCoverEndDateRequest>({
+    params: { dealId, facilityId, amendmentId },
+    session: {
+      user: { ...aPortalSessionUser(), roles: [ROLES.MAKER] },
+      userToken: 'testToken',
+      loginStatus: PORTAL_LOGIN_STATUS.VALID_2FA,
+    },
+  });
+
+const mockDeal = { ...MOCK_BASIC_DEAL, submissionType: DEAL_SUBMISSION_TYPE.AIN, status: DEAL_STATUS.UKEF_ACKNOWLEDGED };
+
+const { req, res } = getHttpMocks();
```

Comment:
@Zainzzkk Do you agree to return that back?

Reply:
`expect(res._getStatusCode()).toEqual(HttpStatusCode.Found);` Line 93 would fail otherwise? @MarRobSoftwire but it's passing in the tests it seems so it seems to be changing from 200?

Reply:
That's because that case uses `res.redirect` & sets the status.

This controller looks to be working correctly so there aren't any examples of this happening, but if there was a bug in the code where it didnt call `return res.render` we might miss it.

Reply:
Please can you revert this change please? @rkontogianni